protocol/signalfx/trace.go

```diff
@@ -522,11 +496,10 @@ type traceErrs struct {
 	lastErr error
 }
 
-// ErrorOrNil returns a non-typed nil error if te is nil or else a non-nil
-// traceErrs instance.
-func (te *traceErrs) ErrorOrNil() error {
+// ToError returns the err object if the it has not been instantiated, and itself if it has
+func (te *traceErrs) ToError(err error) error {
 	if te == nil {
-		return nil
+		return err
 	}
 	return te
```

Comment:
This change (combined with what changed in the `Read` method) will return `err` if there were no previous errors or will return the previous errors without `err`, even when `err` is non-nil.  Is that intended?

Reply:
yes. We need to return the err from the sink unmollested if it exists, but i'd rather return parsing errors otherwise. Maybe i'll update the comment so that is easier to understand.